src/hooks/useGetStreamMetadata.ts

```diff
@@ -24,25 +24,54 @@ export const useGetStreamMetadata = () => {
 		async (streams: string[]) => {
 			if (!userRoles) return;
 			setLoading(true);
+			let encounteredError = false;
```

Comment:
can we change this mutable variable to a state variable

Reply:
This is just a flag to track error state for the whole try/catch. Finally we are setting the state using `setError()` at the end.